packager/packager.cc

```diff
@@ -686,29 +763,37 @@ class ImportTreeBuilder final {
             return lhs.first.show(ctx) < rhs.first.show(ctx);
         });
         for (auto const &[nameRef, child] : childPairs) {
+            // Ignore the entire `Test::*` part of import tree if we are not in a test context.
+            if (moduleType != ImportType::Test && parts.empty() && nameRef == TEST_NAME) {
+                continue;
+            }
             parts.emplace_back(nameRef);
-            makeModule(ctx, child, parts, modRhs, newParentSrc);
+            makeModule(ctx, child, parts, modRhs, moduleType, newParentSrc);
             parts.pop_back();
         }
 
         if (node->source.exists()) {
             if (parentSrc.exists()) {
-                if (auto e = ctx.beginError(node->source.importLoc, core::errors::Packager::ImportConflict)) {
-                    // TODO Fix flaky ordering of errors. This is strange...not being done in parallel,
-                    // and the file processing order is consistent.
-                    e.setHeader("Conflicting import sources for `{}`",
-                                fmt::map_join(parts, "::", [&](const auto &nr) { return nr.show(ctx); }));
-                    e.addErrorLine(core::Loc(ctx.file, parentSrc.importLoc), "Conflict from");
+                // A conflicting import exist. Only report errors while constructing the test output
+                // to avoid duplicate errors because test imports are a superset of normal imports.
+                if (moduleType == ImportType::Test) {
+                    if (auto e = ctx.beginError(node->source.importLoc, core::errors::Packager::ImportConflict)) {
+                        // TODO Fix flaky ordering of errors. This is strange...not being done in parallel,
```

Comment:
Where is this happening / showing up? Is this in a CLI test?

Reply:
Can you confirm via prints that file processing/error emission is actually consistent?

Reply:
it's consistent in a single file and multiple files w/o multithreading hence `--max-threads 0` in https://github.com/sorbet/sorbet/blob/096051e98f8e7128b5506c95aa5bbad645ae930a/test/cli/package-import-conflicts/package-import-conflicts.sh#L3

I believe you said that was ok a while back.